sympy/series/formal.py

```diff
@@ -824,8 +862,12 @@ def _compute_fps(f, x, x0, dir, hyper, order, rational, full):
         return None
 
     ak = sequence(result[0], (k, result[2], oo))
-    xk = sequence(x**k, (k, 0, oo))
-    ind = result[1]
+    if fps_sym:
+        xk = sequence(x**k*x**symb, (k, 0, oo))
```

Comment:
It looks a bit too heuristic. Can we have a more generic approach here?

Reply:
I think that can't be changed. A heuristic approach simplifies things a lot here.